android/libraries/rib-coroutines/src/main/kotlin/com/uber/rib/core/RibDispatchers.kt

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (C) 2022. Uber Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.uber.rib.core
+
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.CoroutineStart
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.IO_PARALLELISM_PROPERTY_NAME
+import kotlinx.coroutines.MainCoroutineDispatcher
+import kotlinx.coroutines.MainScope
+import kotlin.coroutines.ContinuationInterceptor
+
+object RibDispatchers : RibDispatchersProvider {
+  override val Default: CoroutineDispatcher get() = RibCoroutinesConfig.dispatchers.Default
+  override val Main: MainCoroutineDispatcher get() = RibCoroutinesConfig.dispatchers.Main
+  override val IO: CoroutineDispatcher get() = RibCoroutinesConfig.dispatchers.IO
+  override val Unconfined: CoroutineDispatcher get() = RibCoroutinesConfig.dispatchers.Unconfined
+}
+
+data class DefaultRibDispatchers(
```

Comment:
Wondering if we do want this to be a `data class` -- why not a regular class instead?

Reply:
Not strongly opinionated on this one.

Since it's being used to hold dispatchers as an object bag, the equality check *may* be useful by consumers in more complex cases of adapting during testing scenarios or during initialization of an transforming layer from Rx schedulers to Dispatchers.